internal/ipfamily/ipfamily.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package ipfamily // import "go.universe.tf/metallb/internal/ipfamily"
+
+import (
+	"fmt"
+	"net"
+)
+
+// IP family helps identifying single stack IPv4/IPv6 vs Dual-stack ["IPv4", "IPv6"] or ["IPv6", "Ipv4"].
+type IPFamily string
```

Comment:
Nit: just `Family`. From outside we'll call this as ipfamily.IPFamily which is a bit of a repetition.